python/cudf/cudf/core/dtypes.py

```diff
@@ -325,7 +325,7 @@ def serialize(self) -> Tuple[dict, list]:
 
         frames: List[Buffer] = []
 
-        fields = {}
+        fields: Dict[str, Union[bytes, Tuple[Any, Tuple[int, int]]]] = {}
```

Comment:
Could be `Dict[str, Any]` if this kind of verbosity is not preferred.

Reply:
Defining a type alias in `cudf._typing` is another option.

Reply:
The return type from serialize isn't specific enough here, and I don't think this type will appear usefully anywhere else, so I'm inclined just to keep this as is.